holoviews/core/data/__init__.py

```diff
@@ -330,7 +349,7 @@ def __getitem__(self, slices):
         if isinstance(slices, np.ndarray) and slices.dtype.kind == 'b':
             if not len(slices) == len(self):
                 raise IndexError("Boolean index must match length of sliced object")
-            return self.clone(self.interface.select(self, selection_mask=slices))
+            return self.clone(self.select(selection_mask=slices))
```

Comment:
I *think* the old version returned data whereas ``self.select`` should return a ``Dataset``. The former seems more intuitive to me but if they both work it doesn't matter much. Is the new version better somehow?